pkg/artifactory/resource_artifactory_group.go

```diff
@@ -197,7 +198,12 @@ func resourceGroupExists(d *schema.ResourceData, m interface{}) (bool, error) {
 }
 
 func groupExists(client *resty.Client, groupName string) (bool, error) {
-	_, err := client.R().Head(groupsEndpoint + groupName)
+	resp, err := client.R().Head(groupsEndpoint + groupName)
+	if err != nil && resp != nil && resp.StatusCode() == http.StatusNotFound {
```

Comment:
We've seen that, from experience `HEAD` can also return a 400 for this case - please include that code as well